testing.md

```diff
@@ -15,10 +15,56 @@ as well as always iterating on our testing infrastructure.
 
 ## Testing Framework
 
-[`py.test`](https://docs.pytest.org/en/latest/) is the default framework we use
+[`pytest`](https://docs.pytest.org/en/latest/) is the default framework we use
 for testing.
 
 
+## Test Suite Structure
+
+### Test Module Naming/Organization
+
+Where possible, test suites should mirror the module structure of their
+corresponding package.  For example, the
+[`eth-abi`](https://github.com/ethereum/eth-abi) package has a submodule called
+[`eth_abi.grammar`](https://github.com/ethereum/eth-abi/blob/master/eth_abi/grammar.py).
+Accordingly, it has a testing module called
+[`tests.grammar`](https://github.com/ethereum/eth-abi/blob/master/tests/grammar.py)
+that contains testing routines for resources provided by the `eth_abi.grammar`
+module.
+
+Test suites may also include a `tests.end_to_end` module that defines
+functional testing of a package's public API.  This might include common tasks
+performed with a package that compose different individual components of that
+package.
+
+### Test Case Naming
+
+Test case names should adhere to the following format where possible:
+
+```
+def test_<function or class name>_<short description of functionality being tested>:
+    ...
+```
+
+For example, test cases for a function called `foo` might be named as follows:
+
+```python
+def test_foo_does_kung_fu():
+    kung_fu = foo()
+
+def test_foo_raises_value_error():
+    with pytest.raises(ValueError):
+        foo(...)
+
+...
+```
+
+***Note:*** At the time of writing this section on test suite structure, many
```

Comment:
*style nitpick*.  I like the following format for these:

> **Note**: This is a note that somewhat mirrors sphinxes `.. note:` directive.

Reply:
Isn't that an rst thing or does github flavored markdown support that?

Reply:
You can edit his comment to see the source. I'm pretty sure `.md` files are rendered on github the same way comments are.

Reply:
@carver I did check the source.  And I could see that he wasn't using that directive.

Reply:
Ahh, nevermind.  I get what you were saying.  I should be using the `>` line prefix for quotations.